matlab/src/matlab/+arrow/recordbatch.m

```diff
@@ -0,0 +1,57 @@
+%RECORDBATCH Creates an arrow.tabular.RecordBatch from a table.
+
+% Licensed to the Apache Software Foundation (ASF) under one or more
+% contributor license agreements.  See the NOTICE file distributed with
+% this work for additional information regarding copyright ownership.
+% The ASF licenses this file to you under the Apache License, Version
+% 2.0 (the "License"); you may not use this file except in compliance
+% with the License.  You may obtain a copy of the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+% implied.  See the License for the specific language governing
+% permissions and limitations under the License.
+function rb = recordbatch(T)
+    arguments
+        T table
+    end
+
+    arrowArrays = decompose(T);
+    arrayProxyIDs = getArrowProxyIDs(arrowArrays);
+
+    columnNames = string(T.Properties.VariableNames);
+    args = struct(ArrayProxyIDs=arrayProxyIDs, ColumnNames=columnNames);
+    proxyName = "arrow.tabular.proxy.RecordBatch";
+    proxy = arrow.internal.proxy.create(proxyName, args);
+
+    rb = arrow.tabular.RecordBatch(proxy);
+end
+
+function arrowArrays = decompose(T)
```

Comment:
Maybe we can move `decompose` and `getArrowProxyIds` into a package `arrow.tabular.internal` to keep this file a bit smaller?

Reply:
Good point.